lib/api.js

```diff
@@ -0,0 +1,153 @@
+  description: [
+    "The secrets service, typically available at",
+    "`tools.taskcluster.net`, is responsible for managing",
+    "secure data in TaskCluster."
+  ].join('\n')
+});
+
+// Export API
+module.exports = api;
+
+/** Define tasks */
+api.declare({
+  method:      'put',
+  route:       '/name/:name(*)',
+  deferAuth:   true,
+  name:        'set',
+  input:       common.SCHEMA_PREFIX_CONST + "secret.json#",
+  scopes:      [['secrets:set:<name>']],
+  title:       'Create New Secret',
+  description: 'Set a secret associated with some key.'
+}, async function(req, res) {
+    req.satisfies(req.params);
+    let {name} = req.params;
+    let {secret, expires} = req.body;
+    try {
+      await this.entity.create({
+        name:       name,
+        secret:     secret,
+        expires:    new Date(expires)
+      });
+    } catch(e) {
+      res.status(e.statusCode).send(e.code);
+      return;
+    }
+    res.status(204).send();
+});
+
+api.declare({
+  method:      'patch',
+  route:       '/name/:name(*)',
+  deferAuth:   true,
+  name:        'update',
+  input:       common.SCHEMA_PREFIX_CONST + "secret.json#",
+  scopes:      [['secrets:update:<name>']],
+  title:       'Update A Secret',
+  description: 'Update a secret associated with some key.'
+}, async function(req, res) {
+    req.satisfies(req.params);
+    let {name} = req.params;
+    let {secret, expires} = req.body;
+    try {
+      let item = await this.entity.load({name: name});
+      await item.modify(function() {
+        this.secret = secret;
+        this.expires = new Date(expires);
+      });
+    } catch(e) {
+      res.status(e.statusCode).send(e.code);
+      return;
+    }
+    res.status(204).send();
+});
+
+api.declare({
+  method:      'delete',
+  route:       '/name/:name(*)',
+  deferAuth:   true,
+  name:        'remove',
+  scopes:      [['secrets:remove:<name>']],
+  title:       'Delete Secret',
+  description: 'Delete the secret attached to some key.'
+}, async function(req, res) {
+  req.satisfies(req.params);
+  let {name} = req.params;
+  try {
+    await this.entity.remove({name: name});
+  } catch(e) {
+    res.status(e.statusCode).send(e.code);
+    return;
+  }
+  res.status(204).send();
+});
+
+api.declare({
+  method:      'get',
+  route:       '/name/:name(*)',
+  deferAuth:   true,
+  name:        'get',
+  output:      common.SCHEMA_PREFIX_CONST + "secret.json#",
+  scopes:      [['secrets:get:<name>']],
+  title:       'Read Secret',
+  description: 'Read the secret attached to some key.'
+}, async function(req, res) {
+  req.satisfies(req.params);
+  let {name} = req.params;
+  let item = undefined;
+  try {
+    item = await this.entity.load({name: name});
+  } catch(e) {
+    res.status(e.statusCode).send(e.code);
+    return;
+  }
+  if (hasExpired(item.expires)) {
+    res.status(410).send('The requested resource has expired.');
+  } else {
+    res.status(200).json({secret: item.secret, expires: item.expires});
```

Comment:
In this case, before you're not sending an error... use:
`res.reply({secret: item.secret, expires: item.expires})`

Note I would add a method `.json` to `item` will show how in comment on data.js